src/GovUk.Education.ExploreEducationStatistics.Content.Api.Tests/Controllers/DataSetFilesControllerAnalyticsTests.cs

```diff
@@ -0,0 +1,148 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using GovUk.Education.ExploreEducationStatistics.Common;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Fixtures;
+using GovUk.Education.ExploreEducationStatistics.Content.Api.Tests.Fixtures;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Tests.Fixtures;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Tests.Services;
+using Microsoft.AspNetCore.Hosting;
+using Snapshooter.Xunit;
+using Xunit;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Api.Tests.Controllers;
+
+public abstract class DataSetFilesControllerAnalyticsTests : IntegrationTestFixture
+{
+    private readonly DataFixture _fixture = new();
+
+    private DataSetFilesControllerAnalyticsTests(TestApplicationFactory testApp) : base(testApp)
+    {
+    }
+
+    public class DownloadDataSetFileAnalyticsTests(TestApplicationFactory testApp)
+        : DataSetFilesControllerAnalyticsTests(testApp)
+    {
+        public override async Task InitializeAsync() => await StartAzurite();
+
+        [Fact]
+        public async Task DownloadDataSetFile_RecordAnalytics()
+        {
+            Publication publication = _fixture.DefaultPublication()
+                .WithReleases(_ => [_fixture.DefaultRelease(publishedVersions: 1)])
+                .WithTheme(_fixture.DefaultTheme());
+
+            ReleaseFile releaseFile = _fixture.DefaultReleaseFile()
+                .WithReleaseVersion(publication.Releases[0].Versions[0])
+                .WithFile(_fixture.DefaultFile(FileType.Data));
+
+            var testApp = BuildApp(
+                enableAzurite: true,
+                enableAnalytics: true);
```

Comment:
I will try to adopt this into the Public API side of things when I can.

Reply:
It's much nicer to have the generated testApp available at the start of the test - makes everything a lot easier.  It's a shame that the majority of the Public API integration tests only create it within their "action" method.